crates/ty_server/src/server/api/requests/diagnostic.rs

```diff
@@ -105,12 +105,20 @@ fn to_lsp_diagnostic(
         })
         .filter(|mapped_tags| !mapped_tags.is_empty());
 
+    let code = Some(NumberOrString::String(diagnostic.id().to_string()));
     Diagnostic {
         range,
         severity: Some(severity),
         tags,
-        code: Some(NumberOrString::String(diagnostic.id().to_string())),
-        code_description: None,
+        code: code.clone(),
+        code_description: code.and_then(|_| {
+            Some(lsp_types::CodeDescription {
+                href: lsp_types::Url::parse(
+                    format!("https://ty.dev/rules#{}", diagnostic.id()).as_ref(),
+                )
+                .ok()?,
```

Comment:
We shouldn't early return if the URL fails to parse (it would omit the diagnostic). Instead, we should set None (and maybe log a debug message with `tracing::debug`

Reply:
They should. But I'd rather not panic the entire server just because one id isn't a valid anchor

That made me realise. We should only create a link if the id is a `LintName`